### ISSUE TITLE
Tdb ORM Lookups work even on uninitialized record objects

### ISSUE THREAD

**Issue by pixeljunkie** [chameleon-system/chameleon-system#255]
**Describe the bug**
TAdb Record and RecordList classes with Lookup fields have automatically added methods as shortcuts to get records from lookup tables.
example: TAdbShopOrder::GetFieldShopVoucherUseList()
But... it is possible to construct a Tdb object without loading any record. This empty object allows the use of these methods, where now the queries are build with an empty id and may return orphaned records which miss an id (found this issue with archived orders and voucher records)
Of course you should never call lookup or any other method on an empty record, but if you forgot to check for an empty record object this may happen.

**Affected version(s)**
6.2.x

**To Reproduce**
1. temporary undo the fix from #254 
2. open the print order page (see #254)
4. create records in shop_voucher_use table with empty shop_id
5. You should see this record on the print order page

**Expected behavior**
Never run any lookup query with an empty record id.

**Desktop (please complete the following information):**
 - All
**Smartphone (please complete the following information):**
 - All

**Technical details**
Already described above.

**Comment by umulmrum** [comment]
Yes, we should improve this behavior. But the real issue is that there are records without ID and therefore inconsistent data - how is this possible? Did you find a root cause?

**Comment by pixeljunkie** [comment]
> Did you find a root cause?
Yes... and it´s kind of legit. In this case the orders where exported into an archive. But... the shop_voucher_use table is not cleared and this is essential and legt, because if records here would be removed the voucher would be usable again.
Maybe this is not perfect, but lookups need to address this possible case.
